imageBuilder.go

```diff
@@ -9,32 +9,65 @@ import (
 
 var publishImage = flag.Bool("publishImage", false, "Push image to docker hub")
 var codeRunnerConfig = flag.String("codeRunnerConfig", "code-runner.yaml", "Code Runner configuration filename")
+var assignmentEnvConfig = flag.String("assignmentEnvConfig", "assignment-env.yaml", "Assignment Environment configuration filename")
 var dockerAuthConfig = flag.String("dockerAuthConfig", "docker-auth.yaml", "Docker hub authentication configuration filename")
 var dockerfileName = flag.String("dockerfile", "Dockerfile", "Dockerfile name")
 
 func main() {
+
 	flag.Parse()
 
 	log.Println("Creating Dockerfile...")
 
+	imgBuilder := builder.NewImageBuilder(*dockerAuthConfig, *dockerfileName)
+
+	err := generateCodeRunnerImage(imgBuilder)
+	if err == nil {
+		_ = generateAssignmentEnvImage(imgBuilder)
+	}
+
+}
+
+// Generate a dockerfile for code runner server, build its image and push it ot docker hub.
+func generateCodeRunnerImage(imgBuilder *builder.ImageBuilder) error {
+
 	// Unmarshal the yaml configuration file and generate a dockerfileName.
-	err := configurations.WriteDockerfile(*codeRunnerConfig, *dockerfileName)
+	err, _ := configurations.WriteDockerfile(*codeRunnerConfig, *dockerfileName)
```

Comment:
`WriteDockerfile(...)` should not be an associated function of _configurations_.
This function should be part of a utility that uses the configuration to create a Dockerfile.

Reply:
Okay